cloudinit/config/cc_phone_home.py

```diff
@@ -21,7 +21,6 @@
 frequency = PER_INSTANCE
 
 POST_LIST_ALL = [
-    "pub_key_dsa",
```

Comment:
Security review heads up: team behavior changes in stable releases ( 20.04, 22.04) here will only trigger warnings and skip `pub_key_dsa` if provided  in ser-data or system /etc/cloud/cloud-config.*. So images or users how want to report DSA pub_keys within phone_home directives would not have an option to phone home with DSA keys anymore if we backport this change to stable releases. This is probably reasonable behavior for any stable release which still gets updates from upstream as we want to ensure a base-level of security continues to improve. Those who don't want security-related updates should probably during off update pockets such as `jammy-updates`

Reply:
Ack, I think that's fine.  The security benefit is certainly bigger than the loss. If anyone even notices they can just switch their scripts to use RSA.